engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/AnalysisService.java

```diff
@@ -0,0 +1,214 @@
+                    .filter(StringUtils::isNotBlank).collect(Collectors.joining(",")));
+
+            return Optional.ofNullable(executionServices.get(fileExtension)).map(executionService -> {
+                AnalysisRequestStatusDTO dto = executionService.analyze(analysis, analysisDir, callback, updateInterval);
+                if (attachCdmMetadata) {
+                    dto.setExecutionFuture(dto.getExecutionFuture().whenComplete((outcome, throwable) -> {
+                        attachMetadata(analysis, analysisDir);
+                    }));
+                }
+                return dto;
+            }).orElseGet(() -> {
+                log.info("Execution [{}] runtime file type [{}] is not recognized. Skipping", analysis.getId(), fileExtension);
+                return new AnalysisRequestStatusDTO(analysis.getId(), NOT_RECOGNIZED, null, null);
+            });
+
+        } catch (Throwable e) {
+            log.error("Execution [{}] failed to execute", analysis.getId(), e);
+            return new AnalysisRequestStatusDTO(analysis.getId(), NOT_RECOGNIZED, null, null);
+        }
+    }
+
+    public Optional<AnalysisResultDTO> abort(Long id) {
+        return executionServices.values().stream().flatMap(service ->
+                service.getOverseer(id).map(Stream::of).orElseGet(Stream::of)
+        ).map(overseer -> {
+            ExecutionOutcome outcome = abort(id, overseer);
+            return buildResult(id, Date.from(overseer.getStarted()), fromOutcome(outcome));
+        }).findFirst();
+    }
+
+    private ExecutionOutcome abort(Long id, ROverseer overseer) {
+        CompletableFuture<ExecutionOutcome> future = overseer.abort();
+        try {
+            return future.get(30, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            log.info("Execution [{}] waiting for abort operation was interrupted", id);
+            return new ExecutionOutcome(Stage.ABORT, "Waiting for abort interrupted", overseer.getStdout());
+        } catch (ExecutionException e) {
+            log.info("Execution [{}] abort attempt failed", id, e);
+            overseer.getStdout();
+            return new ExecutionOutcome(Stage.ABORT, e.getMessage(), overseer.getStdout() + "\n" + getStackTrace(e.getCause()));
+        } catch (TimeoutException e) {
+            log.info("Execution [{}] waiting for abort timed out", id);
+            return new ExecutionOutcome(Stage.ABORT, e.getMessage(), overseer.getStdout() + "\n" + getStackTrace(e.getCause()));
+        }
+    }
+
+    private void attachMetadata(AnalysisSyncRequestDTO analysis, File analysisDir) {
+        DataSourceUnsecuredDTO dataSource = analysis.getDataSource();
+        try {
+            cdmMetadataService.extractMetadata(dataSource, analysisDir);
+        } catch (Exception e) {
+            log.info("Execution [{}] error collecting CDM metadata for datasource [{}]", analysis.getId(), dataSource.getName(), e);
```

Comment:
The most often case when metadata extraction is failed, is an empty Metadata table. I'd like to suggest not to print full stack trace into the info channel. If it seems useful, then put stacktrace into debug.

Reply:
As far as I tested, on an empty metadata table, analysis will not even start.

Reply:
Out of metadata should not impact analysis. Metadata gathering might be a problem, that's correct. But I'd like to separate analysis issues from metadata issues, and not to see a dozen of useless stacktrace on each analysis submission. Analysis should start regardless of metadata.